Boxed-bolts. #20

lemonboston opened the pull request.

lemonboston commented:
(I had missed to push my last changes, done it now)

dmfs commented:
Please include only stuff that's actually about `Box` and `Boxable` and remove everything that merely uses it, like `FluentBuilder`, `Key` etc.

lemonboston commented:
I've removed all the extra types, so only `Box`, `Boxable`, and `Box` implementations are included now. Squashed and rebased.
Saved the previous state to `icebox/boxed-bolts-extra-stuff` branch.

Ready for check.
One question, the module name is now `boxed-bolts` as in the story - was that intentional or would you maybe like `box-bolts` better?
